server/src/main/java/keywhiz/service/daos/ClientDAO.java

```diff
@@ -135,11 +136,16 @@ public void sawClient(Client client, @Nullable Principal principal) {
     }
   }
 
-  public Optional<Client> getClient(String name) {
+  public Optional<Client> getClientByName(String name) {
     ClientsRecord r = dslContext.fetchOne(CLIENTS, CLIENTS.NAME.eq(name));
     return Optional.ofNullable(r).map(clientMapper::map);
   }
 
+  public Optional<Client> getClientBySpiffeId(String spiffeId) {
+    ClientsRecord r = dslContext.fetchOne(CLIENTS, CLIENTS.SPIFFE_ID.eq(spiffeId));
```

Comment:
the trust domain part should be lower-cased here for comparison

Reply:
Good call--I'll make a commit to handle all SPIFFE IDs as URIs rather than strings (but I'll put it in a new PR so this one doesn't need to be re-reviewed).

Reply:
I don't think this is critical, you could argue it's the client's responsibility to acquire certs that are well formatted.